<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="7" passed="3" failed="4" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2025-07-25T13:39:10 IST" name="Student Name Field Validation Suite" finished-at="2025-07-25T13:40:18 IST" duration-ms="67941">
    <groups>
      <group name="cucumber">
        <method signature="AbstractTestNGCucumberTests.runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:runners.TestRunner@4f18837a]" name="runScenario" class="io.cucumber.testng.AbstractTestNGCucumberTests"/>
      </group> <!-- cucumber -->
    </groups>
    <test started-at="2025-07-25T13:39:10 IST" name="Cucumber-TestNG Execution" finished-at="2025-07-25T13:40:18 IST" duration-ms="67941">
      <class name="runners.TestRunner">
        <test-method is-config="true" signature="setUpClass(org.testng.ITestContext)[pri:0, instance:runners.TestRunner@4f18837a]" started-at="2025-07-25T13:39:10 IST" name="setUpClass" finished-at="2025-07-25T13:39:11 IST" duration-ms="365" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[org.testng.TestRunner@589fb74d]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUpClass -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:runners.TestRunner@4f18837a]" started-at="2025-07-25T13:39:11 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2025-07-25T13:39:22 IST" duration-ms="10819" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA["Student name is empty"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Student Name Field Validation"]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:runners.TestRunner@4f18837a]" started-at="2025-07-25T13:39:22 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2025-07-25T13:39:33 IST" duration-ms="11055" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA["Student name has only 1 character"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Student Name Field Validation"]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Expected message to start with: 'Minimum 2 characters required', but got: 'Registration was successful. The roll number generated is 5510.' expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Expected message to start with: 'Minimum 2 characters required', but got: 'Registration was successful. The roll number generated is 5510.' expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertTrue(Assert.java:57)
at stepdefinitions.RegistrationSteps.validate_dynamic_result(RegistrationSteps.java:97)
at ✽.the result should be "Minimum 2 characters required"(file:///C:/Users/Asus/OneDrive/Desktop/Cogni/sprint2/src/test/resources/features/registration.feature:16)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:runners.TestRunner@4f18837a]" started-at="2025-07-25T13:39:33 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2025-07-25T13:39:42 IST" duration-ms="9049" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA["Student name has 2 characters"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Student Name Field Validation"]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:runners.TestRunner@4f18837a]" started-at="2025-07-25T13:39:42 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2025-07-25T13:39:51 IST" duration-ms="9130" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA["Student name contains numbers"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Student Name Field Validation"]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Expected message to start with: 'No numbers allowed in student name', but got: 'Registration was successful. The roll number generated is 5512.' expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Expected message to start with: 'No numbers allowed in student name', but got: 'Registration was successful. The roll number generated is 5512.' expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertTrue(Assert.java:57)
at stepdefinitions.RegistrationSteps.validate_dynamic_result(RegistrationSteps.java:97)
at ✽.the result should be "No numbers allowed in student name"(file:///C:/Users/Asus/OneDrive/Desktop/Cogni/sprint2/src/test/resources/features/registration.feature:28)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:runners.TestRunner@4f18837a]" started-at="2025-07-25T13:39:51 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2025-07-25T13:40:00 IST" duration-ms="9455" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA["Student name contains special characters"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Student Name Field Validation"]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Expected message to start with: 'No special characters allowed', but got: 'Registration was successful. The roll number generated is 5513.' expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Expected message to start with: 'No special characters allowed', but got: 'Registration was successful. The roll number generated is 5513.' expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertTrue(Assert.java:57)
at stepdefinitions.RegistrationSteps.validate_dynamic_result(RegistrationSteps.java:97)
at ✽.the result should be "No special characters allowed"(file:///C:/Users/Asus/OneDrive/Desktop/Cogni/sprint2/src/test/resources/features/registration.feature:34)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:runners.TestRunner@4f18837a]" started-at="2025-07-25T13:40:00 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2025-07-25T13:40:09 IST" duration-ms="9176" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA["Student name exceeds 30 characters"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Student Name Field Validation"]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Expected message to start with: 'Maximum 30 characters allowed', but got: 'Registration was successful. The roll number generated is 5514.' expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Expected message to start with: 'Maximum 30 characters allowed', but got: 'Registration was successful. The roll number generated is 5514.' expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertTrue(Assert.java:57)
at stepdefinitions.RegistrationSteps.validate_dynamic_result(RegistrationSteps.java:97)
at ✽.the result should be "Maximum 30 characters allowed"(file:///C:/Users/Asus/OneDrive/Desktop/Cogni/sprint2/src/test/resources/features/registration.feature:40)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:runners.TestRunner@4f18837a]" started-at="2025-07-25T13:40:09 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2025-07-25T13:40:18 IST" duration-ms="8825" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA["Student name is valid"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Student Name Field Validation"]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method is-config="true" signature="tearDownClass()[pri:0, instance:runners.TestRunner@4f18837a]" started-at="2025-07-25T13:40:18 IST" name="tearDownClass" finished-at="2025-07-25T13:40:18 IST" duration-ms="27" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDownClass -->
      </class> <!-- runners.TestRunner -->
    </test> <!-- Cucumber-TestNG Execution -->
  </suite> <!-- Student Name Field Validation Suite -->
</testng-results>
